include/seastar/util/log.hh

```diff
@@ -214,20 +214,26 @@ public:
     /// \param args - args to print string
     ///
     template <typename... Args>
+#if FMT_VERSION >= 80000
+    void log(log_level level, fmt::format_string<Args...> fmt, Args&&... args) noexcept {
+#else
     void log(log_level level, format_info fmt, Args&&... args) noexcept {
```

Comment:
The `format_info` maintains the code-location thin on board which is quite useful debugging thing since do_log() may fail to log a message not only because of bad format string. Thus we know which code location failed to tell us something, is it possible to keep this with format-strings?

Reply:
actually, i pondered on alternatives allowing us to preserve code location. but none of them works module a solution to extract the line number of source code path into two template parameters using some meta programming magic. but that would render the code even more obscure .

> do_log() may fail to log a message not only because of bad format string

what could be the possible other cause which throws in the `try` block's `do_log()` call, but makes in the `catch (...)` block?

Reply:
> what could be the possible other cause which throws in the try block's do_log() call, but makes in the catch (...) block?

Memory allocation (to expand the formatters) failures. We've seen it, though I cannot find the respective issue :(

Reply:
https://github.com/scylladb/seastar/issues/980 ?